document-models/real-world-assets/src/_utils/generators.ts

```diff
@@ -26,3 +29,29 @@ export function makeFixedIncomeAssetWithDerivedFields(
 
     return newAsset;
 }
+
+export function makeRwaDocumentWithAssetCurrentValues(
+    document: RealWorldAssetsDocument,
+    currentDate = new Date(),
+) {
+    const documentCopy = copy(document);
+    const transactions = documentCopy.state.global.transactions;
+    const fixedIncomeTypes = documentCopy.state.global.fixedIncomeTypes;
+    const portfolio = documentCopy.state.global.portfolio;
+
+    const portfolioWithCurrentValues = portfolio.map(asset => ({
+        ...asset,
+        ...(isFixedIncomeAsset(asset) && {
+            currentValue: calculateCurrentValue({
```

Comment:
I see that calculateCurrentValue returns a number, is there an issue if we convert the big number to a string instead? It would be safer

Reply:
All of these calculations return numbers right now. Updating it to use strings instead is a future task. I made this one work the same as the others for now, but will definitely change to use strings to store all the values and for this one.